j5basic/Formatters.py

```diff
@@ -25,14 +25,19 @@ class StrftimeFormattedTypeMixIn(object):
     """Mixin for formatting with a Strftime format string."""
 
     def __str__(self):
-        if isinstance(self.format_str, str):
-            # flip through into string world and back again
-            return TimeUtils.strftime(self, self.format_str.encode("UTF-8")).decode("UTF-8")
-        return TimeUtils.strftime(self, self.format_str).decode("UTF-8")
+        format_str = self.format_str
+        if isinstance(format_str, bytes):
+            format_str = format_str.decode('utf-8')
+        fstr = TimeUtils.strftime(self, text_to_native_str(format_str, "utf-8"))
+        if isinstance(fstr, bytes):
+            fstr = fstr.decode("utf-8")
+        return fstr
 
 @python_2_unicode_compatible
 class StrFormattedMixIn(object):
     """Mixin for formatting with a Python % string."""
+    def __init__(self, *args, **kwargs):
+        self._re_entered = False
```

Comment:
Could you remove this now that it's not needed?

Reply:
Done